### ISSUE TITLE
@raw result returning an integer?

### ISSUE THREAD

**Issue by amaravora** [yhirose/maxminddb#42]
Hi! I'm using this library on prod, and noticing that we're getting errors whereby the MaxMindDB::Result we get from performing an i.p. lookup has its  `@raw` instance variable equal to an integer instead of the intended hash.
e.g `=>#<MaxMindDB::Result:0x007f889a1dcb38 @raw=2129834001930874614633>}`

Has anyone ever encountered this before? It seems like it may be a bug. Thanks!

**Comment by amaravora** [comment]
Hi @yhirose no worries, thanks for the library! One example ip is "174.28.1.116" that returned an integer in the 'raw' ivar, but I actually can't reproduce this behavior when I try locally, or even in prod console. I'm starting to wonder whether this is an issue of multiple requests creating some kind of error situation?  Would that be possible?

**Comment by yhirose** [comment]
@amaravora, thanks for the update. I cannot reproduce it either. I tried to make this library thread-safe, so that  'lookup' can be used for multiple request simultaneously. https://github.com/yhirose/maxminddb#regarding-thread-safety
So please let me know if it happens again, since it sounds a pretty bad bug.
Thanks!

**Comment by amaravora** [comment]
@yhirose hey bud, so just for your own benefit, I made a fix to the application code (rails app) such that the problem went away. I had initially created a global instance of the db object returned so that it would live across requests, with the intent being to reduce overhead of establishing the connection to the mmdb for each request. Doing this actually created that strange result of getting integers back in the result hash, so it sounds like the thread-safety implementation you have may not actually be always acting as expected? In any case, after I removed this shared instance, and just let each request instantiate its own instance, the problem went away. 
Let me know if you ever figure this one out, I'm curious myself!
